### DISCUSSION TITLE
Vercel Serverless Function has timed out error (more than 10s to execute)

### DISCUSSION THREAD

**Discussion by yjarid** [vercel/next.js#52768] [Help]
### Summary

I have a next js application deployed on vercel and backend on AWS , I do regular deployement to vercel (once every 2 days on average ) my deployment worked fine until yesterday when I deployed a code with minimal changes mainly styling. I start getting the serverless function Timeout error (it takes more than 10s which is the limit, usualy it takes less than 3s) . so when I rolled back to the old deployment start working fine again . I tried also to re-deploy the same old working version. and surprise surprise I got the serverless function Timeout error again.

I start comparing the build log for the old deployment which working and it 's recent re-deployment. and what I have noticed is the different Vecel Version , in the old one it was "Vercel CLI 31.0.1". and the new one "Vercel CLI 31.0.3" . this is the link to know what was updated between the two versions : [Vercel Release](https://github.com/vercel/vercel/releases)

my getserverSideprops fuction is simple and was working for mor than a year with no issue and no changes. it is still working on the development environment.
```
export async function getServerSideProps({ query, res }) {
  // await avoidRateLimit()

  const apolloClient = intializeApollo()

  let { name, loc,  cat,  neigh, subCat } = query
  
  if (!name ) {
    // would like to save to cache only
    res.setHeader("Cache-Control", "s-maxage=172800, stale-while-revalidate")
  }
 
  let variables = {
    take: 11,
    skip: 0,
    query: name,
  }
 
    if (neigh) { variables.neighborhood = neigh} 
     else {   variables.city = loc  }
  
    if (subCat) {
      variables.subCat = subCat 
    } else if(cat && cat != "-" ){
      variables.cat = cat
    }
  
 let queries = [
    apolloClient.query({ query: GET_BUSINESSES_SEARCH, variables }),
  ]
   await Promise.all(queries)
  
  return {
    props: {
      initialApolloState: apolloClient.cache.extract(),
    },
  }
}
```

this is the main packages : 

```
"node": 18.xx
"next": "^12.2.0",
 "@apollo/client": "^3.7.14",
  "apollo-cache-persist": "^0.1.1",
    "apollo-link-error": "^1.1.12",
    "apollo-link-token-refresh": "^0.3.3",
```

my guess is leaning toward the different Vercel CLI versions , but I am not sure . any help ?

**Comment by icyJoseph** [comment]
Could it be that the application bundle has grown? Have you started to use an icon library or something like that?

It can happen that the compiled commons module grows too big, and that affects the GSSP script parsing time, while on the server.

There's some people who have managed to fight this, I'll try to find a couple of links for you.

**Reply by yjarid** [reply]
yeah , but my confusion is when I tried to re-deploy the last old deployment that is working  I got the time out error as well even though the two deployment are using the same code source . do you still think I still need to investigate common module size issue . or it is something else ?  another information that might help is that I noticed even the `prefetch` request that occurs when you hover over a link that take to the page with the issue `(<Link> `) get timed out as well .

**Reply by icyJoseph** [reply]
Aha, that was not so clear for me from the original post.

If you think this is a Vercel issue, please contact the support over there. 

Are all of your assumptions correct? Maybe the data queries are actually timing out because of some database issue? etc?
